monai/losses/dice.py

```diff
@@ -121,44 +149,42 @@ def forward(self, pred, ground, smooth=1e-5):
         """
         Args:
             pred (tensor): the shape should be BNH[WD].
-            ground (tensor): the shape should be B1H[WD].
+            ground (tensor): the shape should be BNH[WD].
             smooth (float): a small constant to avoid nan.
         """
-        if ground.shape[1] != 1:
-            raise ValueError("Ground truth should have only a single channel, shape is " + str(ground.shape))
-
-        psum = pred.float()
-        if self.do_sigmoid:
-            psum = psum.sigmoid()  # use sigmoid activation
-        if pred.shape[1] == 1:
-            if self.do_softmax:
-                raise ValueError('do_softmax is not compatible with single channel prediction.')
-            if not self.include_background:
+        n_pred_ch = pred.shape[1]
+        if self.do_sigmoid is True:
+            pred = torch.sigmoid(pred)
+        if self.do_softmax is True:
+            pred = torch.softmax(pred, 1)
```

Comment:
as my previous commnet, do_sigmoid=True self.do_softmax=True not compatible. `if self.do_sigmid is True:` could be `if self.do_sigmoid:` (the same for the other places)

Reply:
updated in the latest patch.
Thanks.